iothub/device/src/ConnectionStatusChangeReason.cs

```diff
@@ -8,13 +8,55 @@ namespace Microsoft.Azure.Devices.Client
     /// </summary>
     public enum ConnectionStatusChangeReason
     {
+        /// <summary>
+        /// This signifies that the client is connected, and ready to be used.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Connected"/>.
+        /// </summary>
         Connection_Ok,
+
+        /// <summary>
+        /// This signifies that the SAS token associated with the client has expired, and cannot be renewed.
+        /// The supplied credentials need to be fixed before a connection can be established.
+        /// NOTE: This status change reason is currently not used in the client library.
+        /// </summary>
         Expired_SAS_Token,
+
+        /// <summary>
+        /// This signifies that the device/ module has been deleted or marked as disabled (on your hub instance).
+        /// Fix the device/ module status in Azure before attempting to create the associated client instance.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Disconnected"/>
+        /// </summary>
         Device_Disabled,
+
+        /// <summary>
+        /// This signifies that incorrect credentials were supplied to the client instance.
+        /// The supplied credentials need to be fixed before a connection can be established.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Disconnected"/>
+        /// </summary>
         Bad_Credential,
+
+        /// <summary>
+        /// This signifies that the client was disconnected due to a transient exception, but the retry policy expired before a connection could be re-established.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Disconnected"/>
+        /// </summary>
         Retry_Expired,
+
+        /// <summary>
+        /// This signifies that the client was disconnected due to loss of network.
+        /// NOTE: This status change reason is currently not used in the client library.
+        /// </summary>
         No_Network,
```

Comment:
This status change reason is currently unused in our SDK

Reply:
so what's the reason behind keeping this or the Expired_SAS_Token?

Reply:
Since this is a publicly exposed enum, removing these would break anyone who has referenced these types in their application code (it would still be dead code for them, but the reference would still exist now).

Reply:
But if this is a dead code, they should be updating their application right? How will it help them keeping for example a switch case for this status which is never set/returned by our sdk?
Do we have any documentation/policy which tells us to keep public enums as well which are not used at all? OR any way to deprecate these so can be removed in future?

Reply:
> But if this is a dead code, they should be updating their application right? How will it help them keeping for example a switch case for this status which is never set/returned by our sdk?

This is, and has been dead code for a while now. The "Note:" added will inform them that the SDK will not return this enum value.

> Do we have any documentation/policy which tells us to keep public enums as well which are not used at all? OR any way to deprecate these so can be removed in future?

Ideally, anything that was being used, and we want to deprecate, we would mark them as `[Obsolete]`. However, for this case, this is more of a bug on the SDK, than this feature being deprecated. 
The SDK SHOULD be able to differentiate between sas token invalid vs sas token expired (especially if the service is able to provide different exceptions for both the cases). 
For the "NO_NETWORK" enum, we SHOULD be using it to inform the user that the SDK is retrying, because of a transient network exception.
This is what the java sdk does currently.